### ISSUE TITLE
(interlinks) multiple references error when class as well as containing module have a page

### ISSUE THREAD

**Issue by jmbuhr** [machow/quartodoc#215]
Because of https://github.com/machow/quartodoc/issues/197  (classes in a module are not documented if just `- name: somemodule` is in the contents list),
I tried adding both the module as well as specifically classes in the module like:

```yml
quartodoc:
  package: package
  sidebar: _sidebar.yml
  style: pkgdown
  dir: reference

  sections:
    - title: Modules
      desc: Modules
      contents:
        - name: module
    - title: Things
      desc: Things
      contents:
        - name: module.SomeClass
```

This works to a page for the module and a page for the class, but fails as soon as interlinks e.g.

```
[](`package.module.SomeClass`)
```

are used:

```
Found multiple matches for package.module.SomeClass
1: table: 0x7fe828f3a740
  name: package.module.SomeClass
  role: class
  dispname: -
  domain: py
  priority: 1
  uri: /reference/module.html#package.module.SomeClass
2: table: 0x7fe828f3a740
  name: package.module.SomeClass
  role: class
  dispname: -
  domain: py
  priority: 1
  uri: /reference/module.SomeClass.html#package.module.SomeClass
```

In other words, the quartodoc interlinks filter sort of expects #174 to be implemented.

**Comment by machow** [comment]
Ah, sorry about this. I'm working on supporting class documentation inside modules in https://github.com/machow/quartodoc/pull/212. 

Is the ideal behavior for your case that documenting the module includes the class documentation? If so, #212 should hopefully resolve the issue.

Otherwise, does specifying the members explicitly for the module work for now? E.g.

```
      contents:
        - kind: auto
          name: Builder
          members:

            # specify the exact objects being documented on the module here ----

            - some_method
            - some_other_method
```

**Comment by jmbuhr** [comment]
No worries! I'm just excited to generated documentation with quarto, great project :)

Yes, I think this is what I would intuitively expect.

Weirdly, I have one module in which this does work indeed and one in which it doesn't and I'm still trying to figure out the difference. Though also in the case in which it does work, even though the pages for the classes are generated and accessible via the search, the are not linked anywhere, even with explictly setting `children:` (https://machow.github.io/quartodoc/get-started/basic-content.html#module-and-class-members).

**Comment by jmbuhr** [comment]
I have not been able to fully reproduce the case where classes from one module are rendered and ones from another are not. What I do have now is an example where a **function** is rendered while a **class** is not, both being explicitly listed as members:

https://github.com/jmbuhr/quartodoc-repr


Output: http://jmbuhr.de/quartodoc-repr/reference/module.html#example.module
(shows `main` function but not `MyClass` class, and this time I also can't find it via the search; it is really not rendered.)

**Comment by machow** [comment]
I've released v0.4.0, which has basic support for automatically documenting classes on modules. Does this work for you?

<img width="914" alt="image" src="https://github.com/machow/quartodoc/assets/2574498/5b7ef88b-143b-4f98-b94e-262d7df800df">

What do you think of the tables for each category of thing (e.g. classes, functions)? I'd love to hear how you'd want things laid out on the page :)

**Comment by jmbuhr** [comment]
It's perfect! :)
